truffle/test/03_smart_contract.js

```diff
@@ -60,15 +61,23 @@ describe('SmartContract', async () => {
 
   it("should be allowed to perform a smart contract transaction and gas should be reduced from the balance", async () => {
     const instance = await HelloBlockchainWorld.deployed();
+    const weiBNBalancePreTx = await web3.eth.getBalance(ROOT_ACCOUNT);
 
     try {
       const tx = await instance.incrementHelloCount({
-        from: ROOT_ACCOUNT
+        from: ROOT_ACCOUNT,
+        gas: 100000, // Shouldn't take more than 100 000 to perform a TX
       });
       const txReceipt = await fetchTxReceipt(tx.tx);
+      const weiBNBalancePostTx = await web3.eth.getBalance(ROOT_ACCOUNT);
+      const expectedGasCost = calculateGasCostBN(txReceipt.gasUsed);
```

Comment:
[MINOR] You cannot call this `expected` when you are retrieving the right value from the txReceipt information. That is very confusing. Renamed to just `txGasCost` to avoid future missunderstandings

Reply:
renamed to txGasCost